pallets/ethereum-checked/src/mock.rs

```diff
@@ -0,0 +1,306 @@
+            return CHARLIE_H160;
+        }
+
+        let data = (b"evm:", account_id);
+        return H160::from_slice(&data.using_encoded(blake2_256)[0..20]);
+    }
+}
+
+parameter_types! {
+    pub WeightPerGas: Weight = Weight::from_parts(1, 0);
+    pub const BlockGasLimit: U256 = U256::MAX;
+}
+
+impl pallet_evm::Config for TestRuntime {
+    type FeeCalculator = MockFeeCalculator;
+    type GasWeightMapping = pallet_evm::FixedGasWeightMapping<Self>;
+    type WeightPerGas = WeightPerGas;
+    type BlockHashMapping = pallet_ethereum::EthereumBlockHashMapping<TestRuntime>;
+    type CallOrigin = pallet_evm::EnsureAddressRoot<AccountId>;
+    type WithdrawOrigin = pallet_evm::EnsureAddressTruncated;
+    type AddressMapping = MockAddressMapping;
+    type Currency = Balances;
+    type RuntimeEvent = RuntimeEvent;
+    type Runner = pallet_evm::runner::stack::Runner<Self>;
+    type PrecompilesType = ();
+    type PrecompilesValue = ();
+    type ChainId = ConstU64<1024>;
+    type OnChargeTransaction = ();
+    type BlockGasLimit = BlockGasLimit;
+    type OnCreate = ();
+    type FindAuthor = MockFindAuthor;
+    type Timestamp = Timestamp;
+    type WeightInfo = pallet_evm::weights::SubstrateWeight<TestRuntime>;
+}
+
+parameter_types! {
+    pub const PostBlockAndTxnHashes: PostLogContent = PostLogContent::BlockAndTxnHashes;
+}
+
+impl pallet_ethereum::Config for TestRuntime {
+    type RuntimeEvent = RuntimeEvent;
+    type StateRoot = pallet_ethereum::IntermediateStateRoot<Self>;
+    type PostLogContent = PostBlockAndTxnHashes;
+    type ExtraDataLength = ConstU32<30>;
+}
+
+parameter_types! {
+    pub TxWeightLimit: Weight = Weight::from_parts(u64::max_value(), 0);
+}
+
+impl pallet_ethereum_checked::Config for TestRuntime {
+    type ReservedXcmpWeight = TxWeightLimit;
+    type XvmTxWeightLimit = TxWeightLimit;
+    type InvalidEvmTransactionError = pallet_ethereum::InvalidTransactionWrapper;
+    type ValidatedTransaction = pallet_ethereum::ValidatedTransaction<Self>;
+    type AccountMapping = MockAccountMapping;
+    type TransactOrigin = EnsureXcmEthereumTx<AccountId32>;
+}
+
+pub(crate) type AccountId = AccountId32;
+pub(crate) type BlockNumber = u64;
+pub(crate) type Balance = u128;
+
+pub const ALICE: AccountId32 = AccountId32::new([0u8; 32]);
+pub const BOB: AccountId32 = AccountId32::new([1u8; 32]);
+pub const CHARLIE: AccountId32 = AccountId32::new([2u8; 32]);
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<TestRuntime>;
+type Block = frame_system::mocking::MockBlock<TestRuntime>;
+
+construct_runtime!(
+    pub struct TestRuntime
+    where
+        Block = Block,
+        NodeBlock = Block,
+        UncheckedExtrinsic = UncheckedExtrinsic,
+    {
+        System: frame_system,
+        Timestamp: pallet_timestamp,
+        Balances: pallet_balances,
+        Evm: pallet_evm,
+        Ethereum: pallet_ethereum,
+        EthereumChecked: pallet_ethereum_checked::{Pallet, Call, Origin<T>},
```

Comment:
Do you need to specify `::{Pallet, Call, Origin<T>}` here?

Reply:
It is needed because custom origin is defined in the pallet.

Reply:
I meant it differs from the above declarations - you shouldn't need to specify concrete types/structs.

Reply:
Hmmm, I'm a bit confused. Do you mean it should be consistent with other declarations like `Balances` module?

Reply:
Yes, but as we discussed, it doesn't work so I will mark the comment as resolved.